blocks/columns/columns.js

```diff
@@ -33,23 +33,105 @@ function addSlimPromoClick(block, mq) {
   }
 }
 
+// For children of column <div>, if there is an <a> and a <picture> then move
+// the <picture> to be a child of the <a>, if it isn't already a child.
+function movePictureIntoAnchor(col) {
+  const br = col.querySelector('br');
+  if (br != null) {
+    br.remove();
+  }
+  const aElem = col.querySelector('a[href]');
+  if (aElem != null) {
+    const img = col.querySelector('div > picture > img');
+    if (img != null) {
+      aElem.innerHTML = '';
+      aElem.append(img.parentNode);
+    }
+  }
+}
+
+// For content between <hr> nodes, move all <li> into a single <ul> and move
+// each subsequent <p> as child of the <li>.
+function buildSingleList(col) {
+  const hrList = col.querySelectorAll('hr');
+  if (hrList.length < 1) {
+    return;
+  }
+  const { children } = col;
+  let inBetweenHR = false;
+  let firstUlElem = null;
+
+  [...children].forEach((child) => {
+    if (child.nodeName === 'HR') {
+      inBetweenHR = !inBetweenHR;
+      child.remove();
+      return;
+    }
+
+    if (inBetweenHR) {
+      if (firstUlElem) {
+        if (child.nodeName === 'UL') {
+          // Move the <li> within a <ul> to the first <ul>
+          const liElem = child.querySelector('li');
+          if (liElem) {
+            firstUlElem.append(liElem);
+          }
+        } else {
+          // Move any other element to latest <li>
+          const liElem = Array.from(firstUlElem.querySelectorAll('li')).pop();
+          if (liElem) {
+            liElem.append(child);
+          }
+        }
+      }
+
+      // Remove any <ul> except the first
+      if (child.nodeName === 'UL') {
+        if (firstUlElem == null) {
+          firstUlElem = child;
+        } else {
+          child.remove();
+        }
+      }
+    }
+  });
+}
+
 export default function decorate(block) {
   const cols = [...block.firstElementChild.children];
   block.classList.add(`columns-${cols.length}-cols`);
 
   // setup image columns
   [...block.children].forEach((row) => {
     [...row.children].forEach((col) => {
-      const pic = col.querySelector('picture');
-      if (pic) {
-        const picWrapper = pic.closest('div');
-        if (picWrapper && picWrapper.children.length === 1) {
-          // picture is only content in column
-          picWrapper.classList.add('columns-img-col');
-        }
+      if (col.querySelector('picture')) {
+        // column contains a picture
+        col.classList.add('columns-img-col');
```

Comment:
Like this change to classify the columns based on content (and that you adjusted the selectors in the CSS)